state-chain/pallets/cf-ingress-egress/src/migrations/add_owner_to_channel_details.rs

```diff
@@ -0,0 +1,128 @@
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+
+use crate::*;
+mod old {
+	use super::*;
+
+	#[derive(CloneNoBound, RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+	#[scale_info(skip_type_params(T, I))]
+	pub struct OldDepositChannelDetails<T: Config<I>, I: 'static> {
+		pub deposit_channel: DepositChannel<T::TargetChain>,
+		pub opened_at: TargetChainBlockNumber<T, I>,
+		pub expires_at: TargetChainBlockNumber<T, I>,
+		pub action: ChannelAction<T::AccountId>,
+		pub boost_fee: BasisPoints,
+		pub boost_status: BoostStatus<TargetChainAmount<T, I>>,
+	}
+
+	#[frame_support::storage_alias]
+	pub type DepositChannelLookup<T: Config<I>, I: 'static> = StorageMap<
+		Pallet<T, I>,
+		Twox64Concat,
+		TargetChainAccount<T, I>,
+		OldDepositChannelDetails<T, I>,
+		OptionQuery,
+	>;
+}
+
+pub struct Migration<T: Config<I>, I: 'static>(PhantomData<(T, I)>);
+
+impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for Migration<T, I> {
+	fn on_runtime_upgrade() -> Weight {
+		for (account, channel_details) in old::DepositChannelLookup::<T, I>::drain() {
+			let dummy_account = T::AccountId::decode(&mut &[0u8; 32][..]).unwrap();
```

Comment:
Prefer to use translate rather than drain+insert. According to the docs, inserting during iteration can cause unwanted side effects.